src/compas/geometry/brep/face.py

```diff
@@ -0,0 +1,172 @@
+from compas.data import Data
+
+
+class BrepFace(Data):
```

Comment:
because of how the plugins work, using the default constructor will not trigger one of the backends to be selected.

1. we are okay with this.
2. we overwrite the default constructor, such that it always requires a native face to be provided and creating an empty face is not possible.

Reply:
same for edge, vertex, etc

Reply:
That's a good point. 
I think that `BrepFace`, `BrepVertex`, `BrepEdge` and `BrepLoop` are best described as pure interfaces. Unlike `Brep()` I don't thing they should trigger anything, yet I agree you shouldn't be able to create instances of them. 

Maybe all we need is to define them as abstract?
```python
class BrepFace(metaclass=ABCMeta, Data):
    ...
```

Reply:
if i remember correctly this creates major problem in ironpython with memory leaks etc

Reply:
I see. I've actually heard about this from @gonzalocasas before. Couldn't find any reports that seemed relevant online though. 

Mmh then I guess we can also just leave it for now. Initially, I thought this will cause an issue once we shift to the new COMPAS2.0 serialization, but I'm thinking it might still make sense to do all (de)serialization from `Brep` since it obviously doesn't make sense to mix backends.